### ISSUE TITLE
Autoplay videos from playlist by default

### ISSUE THREAD

**Issue by TippyLion28** [TeamNewPipe/NewPipe#6999]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

At present, in order to start watching videos from a specified point in a playlist, I have to tap "Play All", tap the queue button (and try not to confuse it with the chapters button) and then scroll down in that tiny menu to find the video I want to start from. 

As a user, my expectation is that when I tap a video in a playlist, the next video will be the subsequent video from the playlist, not an autoplay video. 

Could this behaviour be implemented? If not locked behind a setting? 

#### Is your feature request related to a problem? Please describe it

No

#### Additional context

Happy to explain further if unclear 

#### How will you/everyone benefit from this feature?

I will benefit because the tiny queue menu makes me sad.

**Comment by SameenAhnaf** [comment]
Clicking on channel or playlist thumbnail should get the job done, I think.

**Comment by TippyLion28** [comment]
Not sure what you mean? Clicking the thumbnail from inside the playlist is no different from clicking anywhere else on the list item?

**Comment by SameenAhnaf** [comment]
This is what I meant. The same could be applied for channels too.


![IMG_20210828_131855](https://user-images.githubusercontent.com/79245641/131210048-6b5196b6-acf9-419d-bc9f-0cdef803c557.jpg)

**Comment by TippyLion28** [comment]
Perhaps I should've been clearer. I wish to be able to tap a single video inside a playlist, and have it play all subsequent videos from that point onward.

**Comment by SameenAhnaf** [comment]
Then, it's a duplicate of #4805.

**Comment by TippyLion28** [comment]
I would argue it is not a duplicate. #4805 is asking for an explicit context menu item.

I am requesting that this behavior be made the default (or at least, an option in the settings to make it the default.)

From a user's perspective, it's confusing that this is not already the case. When I perform the same action on the YouTube Website, it continues playback from the video I selected. Only when the end of the playlist is reached, does it fall back to autoplay videos.

I assume the YouTube app mirrors this behaviour, but I haven't used the official YouTube app in years, so I can't say for certain.

**Comment by SameenAhnaf** [comment]
Won't #5741 suit you then? I didn't suggest it to you at first as it was a combination of multiple issues.

**Comment by TippyLion28** [comment]
Just for my understanding, your proposed dropdown is solely concerned with the behaviour of the "Play All" button?

I'd argue that if the user can set their preferred behaviour in the settings (including "ask every time"), it would be even better than having a "Play All" button in the first place.

**Comment by SameenAhnaf** [comment]
>An option in settings can be added for "Default Enqeue Method". However, default Enqeue Method drop-down menu should still persist in order to change any time without the hassle of changing settings every time. This option should only mark the enqeue method by default.

(Quoting a part of the issue) You asked for a default option in 'Settings'. Yes, that issue already asks for it.

>user can set their preferred behaviour in the settings (including "ask every time")

That shouldn't be required. The selected 'Default enqueue method' could be marked by default. If any user requires another option, he'll just click on the menu and choose another option.

The only difference from current UI is that 'Play all' button will be no longer available as users can choose 'All' as enqueue method and choose first video to enjoy YouTube like playback.

**Comment by TippyLion28** [comment]
Hmm. From a consistency and user-expectations perspective, I still think it would be better to keep that squirreled away in the settings and have an "ask every time" option, similar to how opening links is handled.

I do think your suggested approach is valid and I would honestly settle for either way, but I will leave this issue open as I feel it is a sufficiently distinct from what you've put forward in #5741

**Comment by opusforlife2** [comment]
From what I've seen, if you play a video from inside a playlist on Youtube, then the playlist opens up in the sidebar, and videos play from it one by one, starting from the one you selected. (I don't remember if the Autoplay (our Auto-enqueue) button's state affects playlist autoplay.)

The same video, when played as a standalone from the Videos tab, does not cause the same behaviour.

Is it possible to add an if-condition where the player checks if the video was opened from a playlist, and enqueues the entire thing?

**Comment by TippyLion28** [comment]
Yeah, that would be ideal. The playlist index would have to be passed through rather than jumping to the first video with a matching URL, in case of duplicates in the playlist. (the official YouTube ui struggles with duplicates, would be nice if we could do it better than them :P )

**Comment by neekt** [comment]
> From what I've seen, if you play a video from inside a playlist on Youtube, then the playlist opens up in the sidebar, and videos play from it one by one, starting from the one you selected.

This is accurate for both the YouTube app on Android and YouTube in the browser. It's also standard playlist behaviour on all major media streaming platforms that I can think of - e.g., SoundCloud, Spotify, Apple Music, Pocket Casts, etc.

I'm inclined to say that the current behaviour - playing a non-playlist video after the user has played a video from within a playlist - is so far removed from "the norm" and user expectation as to be considered a bug.

**Comment by neekt** [comment]
> I do think your suggested approach is valid and I would honestly settle for either way, but I will leave this issue open as I feel it is a sufficiently distinct from what you've put forward in #5741

I also strongly agree with this. I came here specifically to see if this playlist autoplay issue already existed or to create one myself, and the number of tangential issues related to the "Play all" button, enqueueing functionality, "play from here" menu options, etc., made this a little difficult to track down. Please keep this issue open.

**Comment by leberschnitzel** [comment]
I might have a similar problem, but I might misunderstand this problem since I can't even do the workaround:
I have channel groups in subscriptions, for example a soundcloud one with all my soundcloud artists.
When I go to the group, I see all new posts by those artists.
If I click on any song, the following songs in the group are not in the "next" queue, but it's just new suggestions in there.
And I can't find any way to actually play the songs in the order I would expect, since there's no "play all" button.
Is this the same problem or a different one?

**Comment by acrodemocide** [comment]
I've read through the comments on this and several other issues related to auto playing from a playlist. It seems there are some disagreements on the direction the moderators want to go on how the app works, and there were some additional comments indicating that this feature had existed in a previous version but for some reason has not been restored.

There also seems to be some misunderstanding because other issues are often referenced that are related to this but not exactly the same thing. I think the basic idea here is that we would like to be able to open a playlist, click any video in the playlist, then have the playlist continue playing through the rest of the videos in the playlist without any additional prompts or explicitly building a queue and selecting all the following videos in the playlist.

Is anyone actively working on this issue? Is there a reason why anyone would be against someone picking up this issue and creating a pull request for it? I would love to have this functionality, but I also want to be respectful of the process and the main moderators/developers on this project. Please let me know your thoughts.
